hackathon_site/hardware/test_api.py

```diff
@@ -194,3 +194,244 @@ def test_hardware_get_success(self):
         data = response.json()
 
         self.assertEqual(expected_response, data["results"])
+
+
+class HardwareListTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        # Create 3ish hardware objects
+        self.hardware1 = Hardware.objects.create(
+            name="hardware1",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=0,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+
+        self.hardware2 = Hardware.objects.create(
+            name="arduino",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=4,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+
+        self.hardware3 = Hardware.objects.create(
+            name="hardware3",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            notes="notes",
+            quantity_available=4,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+
+        self.view = reverse("api:hardware:hardware-list")
+
+    def _build_filter_url(self, **kwargs):
+        return (
+            self.view + "?" + "&".join([f"{key}={val}" for key, val in kwargs.items()])
+        )
+
+    def test_user_not_logged_in(self):
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_search_by_name(self):
+        self._login()
+
+        url = self._build_filter_url(search="arduino")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        data = response.json()
+        expected_data = {
```

Comment:
Since the actual response format is tested elsewhere in the serializer tests, no need to include the full response body here (because if we change the serializer, we'd have to update this test too). 

I would change the rest of this test case to:

```python
self.assertEqual(len(data["results"]), 1)
self.assertEqual(data["results"][0]["id"], 2)
```

This lets us test the two important things: That only the relevant items are returned (which this test currently misses), and that it is returning the correct items